metaflow/plugins/kubernetes/kubernetes.py

```diff
@@ -189,37 +135,21 @@ def create_job(
         run_time_limit=None,
         env={},
     ):
-        # TODO: Test for DNS-1123 compliance. Python names can have underscores
-        #       which are not valid Kubernetes names. We can potentially make
-        #       the pathspec DNS-1123 compliant by stripping away underscores
-        #       etc. and relying on Kubernetes to attach a suffix to make the
-        #       name unique within a namespace.
-        #
-        # Set the pathspec (along with attempt) as the Kubernetes job name.
-        # Kubernetes job names are supposed to be unique within a Kubernetes
-        # namespace and compliant with DNS-1123. The pathspec (with attempt)
-        # can provide that guarantee, however, for flows launched via AWS Step
-        # Functions (and potentially Argo), we may not get the task_id or the
-        # attempt_id while submitting the job to the Kubernetes cluster. If
-        # that is indeed the case, we can rely on Kubernetes to generate a name
-        # for us.
-        job_name = generate_rfc1123_name(
-            self._flow_name,
-            self._run_id,
-            self._step_name,
-            self._task_id,
-            self._attempt,
-        )
 
         job = (
             KubernetesClient()
             .job(
-                name=job_name,
+                generate_name="t-",
```

Comment:
Obviously not a hard requirement but I found it very useful when those names are somewhat human friendly, when I look at pods or jobs in a K8S cluster management UI (either EKS/K9S/dashboard etc.). 

And I think we do something similar in AWS Batch, where we try to keep them somewhat legible. Maybe we could keep doing the same here?

Reply:
Yep - I wanted to preserve that notion but it wouldn't have been feasible in Argo Workflows - to maintain consistency I ended up switching the behavior for `@kubernetes` as well and stuffed the identifying information in annotations. We can iterate on the pod names post-release as well.